mifospay/src/main/java/org/mifos/mobilewallet/mifospay/savedcards/ui/CardsFragment.kt

```diff
@@ -1,289 +1,68 @@
-            searchView!!.visibility = View.GONE
-            vStateView!!.visibility = View.VISIBLE
-            pbCards!!.visibility = View.GONE
-            val res = resources
-            ivTransactionsStateIcon
-                ?.setImageDrawable(res.getDrawable(R.drawable.ic_cards))
-            tvTransactionsStateTitle?.text = res.getString(R.string.empty_no_cards_title)
-            tvTransactionsStateSubtitle?.text = res.getString(R.string.empty_no_cards_subtitle)
-        }
-    }
-
-    override fun showErrorStateView(drawable: Int, title: Int, subtitle: Int) {
-        pbCards!!.visibility = View.GONE
-        hideSwipeProgress()
-        vStateView!!.visibility = View.VISIBLE
-        if (activity != null) {
-            val res = resources
-            ivTransactionsStateIcon
-                ?.setImageDrawable(res.getDrawable(drawable))
-            tvTransactionsStateTitle?.text = res.getString(title)
-            tvTransactionsStateSubtitle?.text = res.getString(subtitle)
-        }
-    }
-
-    override fun showFetchingProcess() {
-        vStateView!!.visibility = View.GONE
-        searchView!!.visibility = View.GONE
-        rvCards!!.visibility = View.GONE
-        pbCards!!.visibility = View.VISIBLE
-        addCard!!.visibility = View.GONE
-    }
-
-    private fun hideEmptyStateView() {
-        vStateView!!.visibility = View.GONE
-    }
-
-    /**
-     * A function to setup the Layout Manager and Integrate the RecyclerView with Adapter.
-     * This function also implements click action on CardList.
-     */
-    private fun setupCardsRecyclerView() {
-        val llm = LinearLayoutManager(activity)
-        rvCards!!.layoutManager = llm
-        rvCards!!.setHasFixedSize(true)
-        rvCards!!.addItemDecoration(
-            DividerItemDecoration(
-                activity,
-                DividerItemDecoration.VERTICAL
-            )
-        )
-        rvCards!!.adapter = mCardsAdapter
-        rvCards!!.addOnItemTouchListener(
-            RecyclerItemClickListener(
-                activity,
-                object : RecyclerItemClickListener.OnItemClickListener {
-                    override fun onItemClick(childView: View?, position: Int) {
-                        val savedCardMenu = PopupMenu(context, childView)
-                        savedCardMenu.menuInflater.inflate(
-                            R.menu.menu_saved_card,
-                            savedCardMenu.menu
+    ): View {
+
+        //Todo: Getting rid of butterknife after complete FinanceFragment Migration
+        binding = FragmentCardsBinding.inflate(inflater, container, false)
+        ButterKnife.bind(this, binding.root)
+        binding.composeViewCards.apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MifosTheme {
+                    Surface(modifier = Modifier.fillMaxWidth()) {
+                        CardsScreen(
```

Comment:
This is totally wrong implementation, you should handle error UI as composable function, please see how other functionally handling in error code or failed to load. convert into compose and it.
<img width="331" alt="image" src="https://github.com/openMF/mobile-wallet/assets/5724482/a23fdf4d-d179-4d0e-9a1a-d1a0e91d31ae">